packages/salesforcedx-vscode-aura/src/index.ts

```diff
@@ -0,0 +1,124 @@
+    return Uri.parse(value);
+}
+
+export async function activate(context: ExtensionContext) {
+    // UI customizations
+
+    const serverModule = context.asAbsolutePath(path.join('node_modules', 'aura-language-server', 'lib', 'server.js'));
+
+    // The debug options for the server
+    const debugOptions = { execArgv: ['--nolazy', '--inspect=6020'] };
+    // let debugOptions = { };
+
+    // If the extension is launched in debug mode then the debug server options are used
+    // Otherwise the run options are used
+    const serverOptions: ServerOptions = {
+        run: { module: serverModule, transport: TransportKind.ipc },
+        debug: {
+            module: serverModule,
+            transport: TransportKind.ipc,
+            options: debugOptions
+        }
+    };
+
+    const clientOptions: LanguageClientOptions = {
+        outputChannelName: 'Aura Language Server',
+        documentSelector: [
+            {
+                language: 'html',
+                scheme: 'file'
+            },
+            {
+                language: 'html',
+                scheme: 'untitled'
+            },
+            { language: 'javascript', scheme: 'file' },
+            { language: 'javascript', scheme: 'untitled' }
+        ],
+        synchronize: {
+            fileEvents: [
+                workspace.createFileSystemWatcher('**/*.resource'),
+                workspace.createFileSystemWatcher('**/labels/CustomLabels.labels-meta.xml'),
+                workspace.createFileSystemWatcher('**/aura/*/*.{cmp,app,intf,evt,js}'),
+                workspace.createFileSystemWatcher('**/components/*/*/*.{cmp,app,intf,evt,lib,js}'),
+                // need to watch for directory deletions as no events are created for contents or deleted directories
+                workspace.createFileSystemWatcher('**/', true, true, false),
+
+                // these need to be handled because we also maintain a lwc index for interop
+                workspace.createFileSystemWatcher('**/staticresources/*.resource-meta.xml'),
+                workspace.createFileSystemWatcher('**/contentassets/*.asset-meta.xml'),
+                workspace.createFileSystemWatcher('**/lwc/*/*.js'),
+                workspace.createFileSystemWatcher('**/modules/*/*/*.js')
+            ]
+        },
+        uriConverters: {
+            code2Protocol: code2ProtocolConverter,
+            protocol2Code: protocol2CodeConverter
+        }
+    };
+
+    // Create the language client and start the client.
+    this.client = client = new LanguageClient('auraLanguageServer', 'Lightning Language Server', serverOptions, clientOptions);
+    // UI customizations
+    context.subscriptions.push(commands.registerCommand('salesforce-lightning-quickopen', createQuickOpenCommand(client)));
+    const componentProvider = new ComponentTreeProvider(client, context);
+    window.registerTreeDataProvider('salesforce-lightning-components', componentProvider);
+
+    client.onReady().then(() => {
```

Comment:
It would be good to track how long it takes for the server to start, error, etc. We can do it similar to how it's done for the Apex LSP https://github.com/forcedotcom/salesforcedx-vscode/blob/51a10f151ef4328b78642b1cdc587fffd6c6d45f/packages/salesforcedx-vscode-apex/src/index.ts#L87